src/specializations/Tetrahedron.jl

```diff
@@ -27,17 +26,21 @@ end
 #                               Parametric
 ################################################################################
 
-function _parametric(tetrahedron::Meshes.Tetrahedron, t1, t2, t3)
-    if (t3 < 0) || (t3 > 1)
-        msg = "tetrahedron(t1, t2, t3) is not defined for t3 outside [0, 1]."
-        throw(DomainError(t3, msg))
-    end
+# Map argument domain from [0, 1]³ to Barycentric domain for (::Tetrahedron)(t1, t2, t3)
+function _parametric(tetrahedron::Meshes.Tetrahedron)
+    function f(t1, t2, t3)
+        if any(Iterators.map(n -> (n < 0) || (n > 1), (t1, t2, t3)))
+            msg = "tetrahedron(t1, t2, t3) is not defined for (t1, t2, t3) outside [0, 1]³."
+            throw(DomainError((t1, t2, t3), msg))
+        end
 
-    # Take a triangular cross-section at t3
-    a = tetrahedron(t3, 0, 0)
-    b = tetrahedron(0, t3, 0)
-    c = tetrahedron(0, 0, t3)
-    cross_section = Meshes.Triangle(a, b, c)
+        # Take a triangular cross-section at t3
+        a = tetrahedron(t3, 0, 0)
+        b = tetrahedron(0, t3, 0)
+        c = tetrahedron(0, 0, t3)
+        cross_section = _parametric(Meshes.Triangle(a, b, c))
 
-    return _parametric(cross_section, t1, t2)
+        return cross_section(t1, t2)
```

Comment:
[IIUC correctly](https://github.com/JuliaGeometry/MeshIntegrals.jl/pull/139#issuecomment-2510437666) you managed to obtain a significant speedup for `Triangle` by calculating the barycentric coordinates directly from `t1`, `t2`, `t3`. Could we apply a similar trick also for `tetrahedron`, such that we only need to call `tetrahedron` once?

Reply:
The update to `_parametric(::Triangle)` was possible because we only needed to locate a `Point` on a given `Triangle`. Something similar is likely possible for `_parametric(::Tetrahedron)` but would take some time to derive. The simplest path was to locate a triangular cross-section via `t3` and then use `(t1, t2)` as triangle coords, but for optimal performance we'd need to just directly derive 3D-barycentric coords and return something of the form `tetrahedron(a, b, c)`. Definitely worth pursuing, so I documented it in #145 for hopefully near-future work.

Reply:
Nice, thanks! Definitely fine to leave it as is for now.